var/spack/repos/builtin/packages/mpfr/package.py

```diff
@@ -23,11 +24,13 @@ class Mpfr(AutotoolsPackage):
 
     # mpir is a drop-in replacement for gmp
     depends_on('gmp@4.1:')  # 4.2.3 or higher is recommended
-    depends_on('gmp@5.0:', when='@4.0.0:')  # http://www.mpfr.org/mpfr-4.0.0/
+    depends_on('gmp@5.0:', when='@4.0.0:')  # https://www.mpfr.org/mpfr-4.0.0/
 
     # Check the Bugs section of old release pages for patches.
-    # http://www.mpfr.org/mpfr-X.Y.Z/#bugs
+    # https://www.mpfr.org/mpfr-X.Y.Z/#bugs
     patches = {
+        '4.0.2': 'f2d2a530acb5e70e1a9d5b80881dbb4a504d56535c4bc103d83e0bb630172029',
```

Comment:
I'm hesitant about adding this patch, as its hash will likely change as more bugs are discovered. MPFR has 2 types of patches available, `allpatches` which contains all the patches for known bugs in a particular release, and individual patches. We would replace the `allpatches` with individual patches, but then there will be like a dozen patches per version.